pkg/controller/iotconfig/infinispan_device_connection.go

```diff
@@ -0,0 +1,239 @@
+
+		if service.Infinispan != nil {
+			applyContainerConfig(container, service.Infinispan.Container)
+		}
+
+		// apply infinispan server options
+
+		if service.Infinispan.Server.External != nil {
+			if err := appendInfinispanExternalConnectionServer(container, config.Spec.ServicesConfig.DeviceConnection.Infinispan.Server.External); err != nil {
+				return err
+			}
+		} else {
+			return util.NewConfigurationError("infinispan backend server configuration missing")
+		}
+
+		// return
+
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	// reset init containers
+
+	deployment.Spec.Template.Spec.InitContainers = nil
+
+	// tracing
+
+	SetupTracing(config, deployment, tracingContainer)
+
+	// volumes
+
+	install.ApplyConfigMapVolume(&deployment.Spec.Template.Spec, "config", nameDeviceConnection+"-config")
+
+	// inter service secrets
+
+	if err := ApplyInterServiceForDeployment(config, deployment, nameDeviceConnection); err != nil {
+		return err
+	}
+
+	// return
+
+	return nil
+}
+
+func appendInfinispanExternalConnectionServer(container *v1.Container, external *iotv1alpha1.ExternalInfinispanDeviceConnectionServer) error {
+
+	// basic connection
+
+	install.ApplyEnvSimple(container, "ENMASSE_IOT_DEVICECONNECTION_INFINISPAN_HOST", external.Host)
+	install.ApplyEnvSimple(container, "ENMASSE_IOT_DEVICECONNECTION_INFINISPAN_PORT", strconv.Itoa(int(external.Port)))
+	install.ApplyEnvSimple(container, "ENMASSE_IOT_DEVICECONNECTION_INFINISPAN_USERNAME", external.Username)
+	install.ApplyEnvSimple(container, "ENMASSE_IOT_DEVICECONNECTION_INFINISPAN_PASSWORD", external.Password)
+
+	// SASL
+
+	install.ApplyOrRemoveEnvSimple(container, "ENMASSE_IOT_DEVICECONNECTION_INFINISPAN_SASLSERVERNAME", external.SaslServerName)
+	install.ApplyOrRemoveEnvSimple(container, "ENMASSE_IOT_DEVICECONNECTION_INFINISPAN_SASLREALM", external.SaslRealm)
+
+	// cache names
+
+	deviceStates := ""
+	if external.CacheNames != nil {
+		deviceStates = external.CacheNames.DeviceConnections
+	}
+
+	install.ApplyOrRemoveEnvSimple(container, "ENMASSE_IOT_DEVICECONNECTION_INFINISPAN_CACHENAMES_DEVICES", deviceStates)
+
+	// done
+
+	return nil
+}
+
+func (r *ReconcileIoTConfig) reconcileInfinispanDeviceConnectionConfigMap(_ *iotv1alpha1.IoTConfig, configMap *corev1.ConfigMap) error {
+
+	install.ApplyDefaultLabels(&configMap.ObjectMeta, "iot", configMap.Name)
+
+	if configMap.Data == nil {
+		configMap.Data = make(map[string]string)
+	}
+
+	if configMap.Data["logback-spring.xml"] == "" {
+		configMap.Data["logback-spring.xml"] = DefaultLogbackConfig
+	}
+
+	configMap.Data["application.yml"] = `
+hono:
+
+  auth:
+    port: 5671
+    keyPath: /etc/tls/tls.key
+    certPath: /etc/tls/tls.crt
+    keyFormat: PEM
+    trustStorePath: /var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt
+    trustStoreFormat: PEM
+
+enmasse:
+  iot:
+
+    app:
+      maxInstances: 1
+
+    vertx:
+      preferNative: true
+
+    healthCheck:
+      insecurePortBindAddress: 0.0.0.0
```

Comment:
Should this be 127.0.0.1?

Reply:
This needs to be accessed by the health probe and the prometheus scraper. From what I remember that didn't work when it was bound to 127.0.0.1.